packages/date/buildLocalizeFn.d.ts

```diff
@@ -0,0 +1,6 @@
+declare module 'date-fns/locale/_lib/buildLocalizeFn' {
+  type LocalizeFunction = (...args: any[]) => any;
+  type LocalizeFunctionBuilder = (args: any) => LocalizeFunction;
+  const localizeFunctionBuilder: LocalizeFunctionBuilder;
+  export default localizeFunctionBuilder;
+}
```

Comment:
@swissspidy, is this best-practice? Not really sure, but it seems to work and is about as "well-typed" as the regular `localize` function from that package.

Reply:
Maybe we want this file in a different location? Or with a different filename?

Reply:
That is a great question. I don't know anymore why / how I ended up using this function but apparently it's still not public and thus has no types or anything (see https://github.com/date-fns/date-fns/issues/1116)

So this seems like a reasonable workaround.

In other packages I've put this kinda stuff in a dedicated `typings` folder, which should work here as well. E.g. `packages/date/src/typings/date-fns/index.d.ts`. But open to other suggestions.

Reply:
Moved in b07750287e.